From 2fe76487f12078b8f2d474148798597cdfb63ea2 Mon Sep 17 00:00:00 2001
From: Joshua Watt <JPEWhacker@gmail.com>
Date: Mon, 27 Feb 2023 21:31:45 -0600
Subject: [PATCH] cmake: Install files

Add an install step for the files that zdoom creates

Upstream-Status: Inactive-Upstream [lastrelease: 2016]
Signed-off-by: Joshua Watt <JPEWhacker@gmail.com>
---
 src/CMakeLists.txt    | 2 ++
 wadsrc/CMakeLists.txt | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0c0955b..7ae60d0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -9,6 +9,7 @@ include( CheckFunctionExists )
 include( CheckCXXCompilerFlag )
 include( CheckLibraryExists )
 include( FindPkgConfig )
+include( GNUInstallDirs )
 
 if( NOT APPLE )
 	option( NO_ASM "Disable assembly code" OFF )
@@ -1172,6 +1173,7 @@ add_executable( zdoom WIN32 MACOSX_BUNDLE
 	r_data/r_translate.cpp
 	zzautozend.cpp
 )
+install(TARGETS zdoom DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 set_source_files_properties( xlat/parse_xlat.cpp PROPERTIES OBJECT_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/xlat_parser.c" )
 set_source_files_properties( sc_man.cpp PROPERTIES OBJECT_DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/sc_man_scanner.h" )
diff --git a/wadsrc/CMakeLists.txt b/wadsrc/CMakeLists.txt
index 0ecb3d5..4bf33f0 100644
--- a/wadsrc/CMakeLists.txt
+++ b/wadsrc/CMakeLists.txt
@@ -1,3 +1,6 @@
 cmake_minimum_required( VERSION 2.4 )
 
+include( GNUInstallDirs )
+
 add_pk3(zdoom.pk3 ${CMAKE_CURRENT_SOURCE_DIR}/static)
+install(FILES ${CMAKE_BINARY_DIR}/zdoom.pk3 DESTINATION ${CMAKE_INSTALL_BINDIR})
